### ISSUE TITLE
String enum values not recognized (false positive invalid-str-returned)

### ISSUE THREAD

**Issue by ppentchev** [PyCQA/pylint#3649]
### Steps to reproduce
1. Run pylint on the attached xenum.py file.
2. Wonder why it says that the returned value is not a string even for the classes that explicitly subclass "str" before "Enum" and regardless of the isinstance assertion that's good enough for mypy. (now why would mypy need the assertion even for the classes that explicitly subclass str before Enum is a completely different topic for a completely different venue)

### Current behavior

`E0307: __str__ does not return str (invalid-str-returned)` reported for all four methods.

### Expected behavior

Nothing reported at all.

### pylint --version output
pylint 2.5.2
astroid 2.4.1
Python 3.8.3 (default, May 26 2020, 08:47:10) 
[GCC 9.3.0]
[xenum-py.txt](https://github.com/PyCQA/pylint/files/4687019/xenum-py.txt)

**Comment by Snawoot** [comment]
I also have this issue. Code:

```python3
class LogLevel(enum.IntEnum):
    debug = logging.DEBUG
    info = logging.INFO
    warn = logging.WARN
    error = logging.ERROR
    fatal = logging.FATAL
    crit = logging.CRITICAL

    def __str__(self):
        return self.name
```

**Comment by Pierre-Sassoulas** [comment]
Closing because it's been fixed in the main branch.
